### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by codegrue** [cydergoth/tembed-app-shell#2]
Thank you for this demo. I'm trying to replicate your screen capture code. But I noticed your LOCK_GUI macros. Why do you use xSemaphoreTakeRecursive instead of xSemaphoreTake?

**Comment by codegrue** [comment]
I am going to have to modify your snap code a bit becuase I am out of main heap and need to use PSRAM for the snapshot. I'll post a solution if I get one.

**Comment by cydergoth** [comment]
Yes, you do have to chose which functions are most critical to your use
case with small memory devices like the ESP32. Good luck, let me know how
it goes!
